mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/CassandraSmtpTestRule.java

```diff
@@ -29,41 +30,67 @@
 import org.apache.james.dnsservice.api.InMemoryDNSService;
 import org.apache.james.modules.protocols.ProtocolHandlerModule;
 import org.apache.james.modules.protocols.SMTPServerModule;
+import org.apache.james.modules.protocols.SmtpGuiceProbe;
 import org.apache.james.modules.server.CamelMailetContainerModule;
+import org.apache.james.mpt.api.Continuation;
+import org.apache.james.mpt.api.Session;
 import org.apache.james.mpt.monitor.SystemLoggingMonitor;
 import org.apache.james.mpt.session.ExternalSessionFactory;
-import org.apache.james.mpt.smtp.SmtpHostSystem;
 import org.apache.james.queue.api.MailQueueItemDecoratorFactory;
 import org.apache.james.queue.api.RawMailQueueItemDecoratorFactory;
 import org.apache.james.server.core.configuration.Configuration;
 import org.apache.james.util.Host;
 import org.apache.james.util.Port;
 import org.apache.james.utils.DataProbeImpl;
 import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
 
 import com.google.common.base.Preconditions;
 import com.google.common.base.Splitter;
 import com.google.inject.Module;
 import com.google.inject.util.Modules;
 
-public class CassandraJamesSmtpHostSystem extends ExternalSessionFactory implements SmtpHostSystem {
+public class CassandraSmtpTestRule implements TestRule, SmtpHostSystem {
+
+    enum SmtpServerConnectedType {
+        SMTP_GLOBAL_SERVER(SmtpGuiceProbe::getSmtpPort),
+        SMTP_STAR_TLS_SERVER(SmtpGuiceProbe::getSmtpsPort);
+
+        private final Function<SmtpGuiceProbe, Integer> portExtractor;
+
+        private SmtpServerConnectedType(Function<SmtpGuiceProbe, Integer> portExtractor) {
+            this.portExtractor = portExtractor;
+        }
+
+        public Function<SmtpGuiceProbe, Integer> getPortExtractor() {
```

Comment:
too bad that the port extractor does not extract a port but an Integer (:

Reply:
I think what @mbaechler  wants is to rather use the `Port` object (strong typing)